src/mbgl/renderer/render_layer.cpp

```diff
@@ -71,8 +76,60 @@ bool RenderLayer::needsRendering(float zoom) const {
            && baseImpl->maxZoom >= zoom;
 }
 
-void RenderLayer::setRenderTiles(std::vector<std::reference_wrapper<RenderTile>> tiles) {
-    renderTiles = std::move(tiles);
+void RenderLayer::setRenderTiles(RenderTiles tiles, const TransformState& state) {
+    renderTiles = filterRenderTiles(std::move(tiles));
+    sortRenderTiles(state);
+}
+
+RenderLayer::RenderTiles RenderLayer::filterRenderTiles(RenderTiles tiles) const {
+    return filterRenderTiles(std::move(tiles), [](const auto& tile){ return !tile.tile.isRenderable() && tile.tile.holdForFade(); });
```

Comment:
This doesn't look logically equivalent to previous behavior -- if `holdForFade()` is false then the condition will always be false, the filter will always pass, and the tile will always render, even if `isRenderable()` is false.

Reply:
Thanks, fixed.